content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,652 @@
+if\ \text{confidence} > c_{target} & THEN \ \text{finalize decision} \newline
+\end{array}
+$$
+
+After the `OPINION` phase is executed, the current value of `confidence`
+is considered: if `confidence` exceeds a threshold derived from the
+network size and directly related to the total votes received, an
+honest node marks the decision as final, and always returns this
+opinion is response to further queries from other nodes on the
+network.
+ 
+    IF 
+      confidence > confidence_threshold
+    OR 
+      round > max_rounds
+    THEN
+      finalized <-- T
+      QUERY LOOP TERMINATES
+    ELSE 
+      round +== 1
+      QUERY LOOP CONTINUES
+
+Thus, after the decision phase, either a decision has been finalized
+and the local node becomes quiescent never initiating a new query, or
+it initiates a [new query](#query).
+
+### Termination
+
+A local round of Claro terminates in one of the following following
+execution model considerations:
+
+
+1.  No queries are received for any newly initiated round for temporal
+    periods observed via a locally computed passage of time.  See [the
+    following point on local time](#clock).
+
+2.  The `confidence` on the proposal exceeds our threshold for
+    finalization.
+    
+3.  The number of `rounds` executed would be greater than
+    `max_rounds`. 
+    
+#### Quiescence
+
+After a local node has finalized an `opinion` into a `decision`, it enters a quiescent
+state whereby it never solicits new votes on the proposal.  The local
+node MUST reply with the currently finalized `decision`.
+
+#### Clock
+
+The algorithm only requires that nodes have computed the drift of
+observation of the passage of local time, not that that they have
+coordinated an absolute time with their peers.  For an implementation
+of a phase locked-loop feedback to measure local clock drift see
+[NTP](https://www.rfc-editor.org/rfc/rfc5905.html).
+
+## Further points
+
+### Node receives information during round
+In the query step, the node is envisioned as packing information into
+the query to cut down on the communication overhead a query to each of
+this `k` nodes containing the node's own current opinion on the
+proposal (`YES`, `NO`, or `NONE`).  The algorithm does not currently
+specify how a given node utilizes this incoming information.  A
+possible use may be to count unsolicited votes towards a currently
+active round, and discard the information if the node is in a
+quiescent state.
+
+#### Problems with Weighting Node Value of Opinions
+If the view of other nodes is incomplete, then the sum of the optional
+weighting will not be a probability distribution normalized to 1.
+
+The current algorithm doesn't describe how the initial opinions are formed.
+
+# Implementation status
+The following implementations have been created for various testing and simulation purposes:
+- [Rust](https://github.com/logos-co/consensus-research)
+- [Python]() - FILL THIS IN WITH NEWLY CREATED REPO
```

Comment:
should we just publish the `consensus-models` which is currently private?

Reply:
you mentioned you wanted to publish each implementation into a separate repo. I'd say each implementation should be able to be used as a library to other people and not clumped together.